api/v1beta1/index/machine.go

```diff
@@ -86,10 +85,5 @@ func machineByProviderID(o client.Object) []string {
 		return nil
 	}
 
-	providerID, err := noderefutil.NewProviderID(*machine.Spec.ProviderID)
-	if err != nil {
-		// Failed to create providerID, skipping.
-		return nil
-	}
-	return []string{providerID.IndexKey()}
+	return []string{*machine.Spec.ProviderID}
```

Comment:
Could we assign the value on line 85 from `StringDeref` call, and reuse it here?

Reply:
yes, done!